projects/ripe/index.js

```diff
@@ -0,0 +1,164 @@
+
+const ONE_HUNDRED = BigInt(10000)
+const RIPE_MAX_SUPPLY = BigInt(1000000000) * BigInt(10) ** BigInt(18)
+
+async function getGreenLpValues(api, address){
+  const [usdcBalance, greenBalance, greenLpSupply, addressLp] = await Promise.all([
+    api.call({
+      abi: 'erc20:balanceOf',
+      target: USDC_CONTRACT,
+      params: [GREEN_LP_CONTRACT],
+    }),
+    api.call({
+      abi: 'erc20:balanceOf',
+      target: GREEN_CONTRACT,
+      params: [GREEN_LP_CONTRACT],
+    }),
+    api.call({
+      abi: 'erc20:totalSupply',
+      target: GREEN_LP_CONTRACT,
+    }),
+    api.call({
+      abi: 'erc20:balanceOf',
+      target: GREEN_LP_CONTRACT,
+      params: [address]
+    }),
+  ])
+  
+  const usdc = BigInt(usdcBalance) 
+  const green = BigInt(greenBalance)
+  const lpAddressShare = (BigInt(addressLp) * ONE_HUNDRED) / BigInt(greenLpSupply)
+  
+  const usdcFinal = (lpAddressShare * usdc) / ONE_HUNDRED 
+  const greenFinal = (lpAddressShare * green) / ONE_HUNDRED
+
+  api.add(USDC_CONTRACT, usdcFinal)
+  api.add(GREEN_CONTRACT, greenFinal)
+}
+
+async function getPairs(){
+  const response = await fetch('https://api.ripe.finance/api/ripe/assets');
+  const assets = await response.json();
+  const assetsArray = assets.result.filter(a => a.tokenAddress.toLowerCase() !== SGREEN_CONTRACT.toLowerCase() && a.tokenAddress.toLowerCase() !== GREEN_LP_CONTRACT.toLowerCase()); 
+  const stabilityPoolAddress = assets.result.find(a => a.vaultId === 1).vaultAddress
+  const nonSpAssets = assetsArray.filter(a => a.vaultId > 2)
+
+  const assetVaultPairs = [
+    ...nonSpAssets.map(({ tokenAddress, vaultAddress }) => [tokenAddress, vaultAddress]), 
+    ...nonSpAssets.map(({ tokenAddress }) => [tokenAddress, stabilityPoolAddress]),
+    [GREEN_CONTRACT, SGREEN_CONTRACT],
+    [USDC_CONTRACT, ENDAOMENT_CONTRACT],
+    [GREEN_CONTRACT, ENDAOMENT_CONTRACT]
+  ];
+
+  return {assetVaultPairs, stabilityPoolAddress};
+}
+
+async function tvl(api) {
+  const { assetVaultPairs: pairs, stabilityPoolAddress } = await getPairs();
+
+  const calls = pairs.map(([token, vault]) => {
+    return api.call({
+      abi: 'erc20:balanceOf',
+      target: token,
+      params: [vault],
+    });
+  });
+
+  const balances = await Promise.all(calls);
+  
+  const underlyingCalls = pairs.map(([token], index) => {
+    return api.call({
+      abi: "function getUnderlyingData(address, uint256) view returns (tuple(address asset, uint256 amount, uint256 usdValue, uint256 legId, address legoAddr, string legoDesc))",
+      target: LEGO_TOOLS_CONTRACT,
+      params: [token, balances[index]],
+        
+    })
+  })
+
+  const underlyingBalances = await Promise.all(underlyingCalls);
+  underlyingBalances.forEach(({asset, amount}) => {
+    if(amount !== '0'){
+        api.add(asset, amount);
+    }
+  });
+
+  await getGreenLpValues(api, stabilityPoolAddress)
+  await getGreenLpValues(api, ENDAOMENT_CONTRACT)
+}
+
+async function vesting(api){
```

Comment:
vesting export is more for token streaming services etc where they facilitate vesting for third parties - I think we should remove it here (if I understand the project correctly)